### PR TITLE
[3928] Reactify History dialog

### PR DESCRIPTION
**Pull request by aachavarria**
GeneralDialogs(https://github.com/craftercms/craftercms/issues/3976): changing Dialog onCancel to onDismiss;
HistoryDialog(https://github.com/craftercms/craftercms/issues/3928):
-Use API2 for config files //
-Show 10 instead of 20 history items //
-Clicking a row outside of "compare to" mode, should "view” item //
-Compare to current is not implemented //
-"Select a revision to compare" state should not show item menu //

### CODE REVIEW

**Review by rart — CHANGES_REQUESTED** [review]
common api missing format on new lines

File: ui/app/src/modules/Content/Delete/DeleteDialog.tsx [hunk 1]
```diff
@@ -296,7 +299,7 @@ function DeleteDialog(props: DeleteDialogProps) {
     deleteItems(siteId, user.username, submissionComment, data).subscribe(
       (response) => {
         setApiState({ submitting: false });
-        onSuccess?.(response);
+        onSuccess(response);
```
Comment (rart) [inline ui/app/src/modules/Content/Delete/DeleteDialog.tsx]: Prop is optional. Leave ? as it was

File: ui/app/src/modules/Content/Dependencies/DependencySelection.tsx [hunk 2]
```diff
@@ -291,7 +291,7 @@ export function DependencySelectionDelete(props: DependencySelectionDeleteProps)
       .filter(([key, value]) => value === true)
       .map(([key]) => key);
     onChange(result);
-  }, [checked, onChange]);
+  }, [checked]);
```
Comment (rart) [inline ui/app/src/modules/Content/Dependencies/DependencySelection.tsx:296]: Removed dependency. If dep creates loop, useCallback where needed

File: ui/app/src/utils/codebase-bridge.tsx [hunk 3]
```diff
@@ -274,10 +275,17 @@ export function createCodebaseBridge() {
           reject(e);
         }
       });
+    },
+    createLegacyCallbackListener(id: string, listener: any) {
```
Comment (rart) [inline ui/app/src/utils/codebase-bridge.tsx]: Fix types

File: ui/app/src/utils/codebase-bridge.tsx [hunk 4]
```diff
@@ -274,10 +275,17 @@ export function createCodebaseBridge() {
...
+        listener();
+        document.removeEventListener(id, listener, false);
+      };
+      document.addEventListener(id, callback, true);
     }
 
   };
 
-  // @ts-ignore
+// @ts-ignore
```
Comment (rart) [inline ui/app/src/utils/codebase-bridge.tsx]: Restore spacing

File: ui/app/src/modules/Content/Dependencies/DependencySelection.tsx (hunk 2 above)
Comment (aachavarria) [reply ui/app/src/modules/Content/Dependencies/DependencySelection.tsx:296]: Jose will fix this issue

**Comment by aachavarria** [comment]
build tested successful

File: ui/app/src/modules/Content/History/HistoryDialog.tsx [hunk 5]
```diff
@@ -403,12 +420,13 @@ export default function HistoryDialog(props: HistoryDialogProps) {
 
   const [menu, setMenu] = useSpreadState<Menu>(menuInitialState);
 
-  const rowsPerPage = 20;
+  const rowsPerPage = 10;
```
Comment (rart) [inline ui/app/src/modules/Content/History/HistoryDialog.tsx:423]: Let's make this a prop with default to 10

**Review by rart — APPROVED** [review]

**Pull request merged**